tests/test_sync.py

```diff
@@ -92,105 +94,122 @@ def test_local_equals_API(self, pir_to_xy_response, downsample_ref):
-    @pytest.mark.parametrize("downsample_ref", [25, 50])
-    @pytest.mark.parametrize("axis", ["coronal", "sagittal"])
-    def test_patched(self, include_expression, downsample_ref, axis, monkeypatch):
-        """Does not requires internet, everything is patched.
-
-        The only thing that is unpatched is the `get_transform_simple`.
-        """
+    def test_patched(self, include_expression, data_folder, monkeypatch):
+        """Does not requires internet, everything is patched."""
 
         # Parameters
-        dataset_id = 12345
+        dataset_id = 123  # Sagittal dataset
+        downsample_ref = 25
+        grid_shape = (13200 // downsample_ref, 8000 // downsample_ref)
 
         # Mocking
-        get_2d_bulk_fake = Mock(
-            return_value={
-                11111: (np.ones((2, 3)), 20),
-                22222: (np.ones((2, 3)), 50),
-            }
-        )
-
-        get_3d_fake = Mock(
-            return_value=np.ones((3, 4)),
-        )
-
-        common_queries_fake = Mock()
-        common_queries_fake.get_axis.return_value = axis
-
-        get_image_fake = Mock(
-            return_value=np.zeros((10, 10)),
-        )
-
-        xy_to_pir_fake = Mock(return_value=(1200, 235, 242))
-
-        # Patching
-        monkeypatch.setattr("atldld.sync.get_2d_bulk", get_2d_bulk_fake)
-        monkeypatch.setattr("atldld.sync.get_3d", get_3d_fake)
-        monkeypatch.setattr("atldld.sync.get_image", get_image_fake)
-        monkeypatch.setattr("atldld.sync.CommonQueries", common_queries_fake)
-        monkeypatch.setattr("atldld.sync.xy_to_pir_API_single", xy_to_pir_fake)
+        def mock_rma_all(value):
+            parameters_dataset = RMAParameters(
+                model="SectionDataSet",
+                criteria={
+                    "id": dataset_id,
+                },
+                include=["alignment3d"],
+            )
+            parameters_images = RMAParameters(
+                model="SectionImage",
+                criteria={
+                    "data_set_id": dataset_id,
+                },
+                include=["alignment2d"],
+            )
+            if value == parameters_dataset:
+                with open(data_folder / "rma_all" / "SectionDataSet.json") as f:
+                    data = json.load(f)
+                return [
+                    data,
+                    None,
+                ]  # We extract only the first element of the list here.
+            elif value == parameters_images:
+                with open(data_folder / "rma_all" / "SectionImage.json") as f:
+                    data = json.load(f)
+                return [
+                    data,
+                ]  # In a list because can have several images
+            else:
+                raise ValueError("Not expected")
+
+        mock = MagicMock(side_effect=mock_rma_all)
+        monkeypatch.setattr("atldld.sync.rma_all", mock)
+        monkeypatch.setattr("atldld.utils.get_image", np.zeros(grid_shape))
```

Comment:
`monkeypatch.setattr("atldld.utils.get_image", np.zeros(grid_shape))` does not work (at least on my computer). I think one always needs to specify the module that imports the function (`sync`) rather than the one where it is implemented (`utils`).

```suggestion
        fake_rma_all = Mock(side_effect=mock_rma_all)
        fake_get_image = Mock(return_value=np.zeros(grid_shape))
        monkeypatch.setattr("atldld.sync.rma_all", fake_rma_all)
        monkeypatch.setattr("atldld.sync.get_image", fake_get_image)
```

I also change the variable naming since there are 2 mocks

Reply:
Done in 48d4b15. Thanks a lot for catching this!